frontend/src/metabase/containers/SaveQuestionModal/SaveQuestionModal.tsx

```diff
@@ -217,7 +217,12 @@ export const SaveQuestionModal = ({
     [originalQuestion, handleOverwrite, handleCreate],
   );
 
-  const isSavedQuestionChanged = useSelector(getIsSavedQuestionChanged);
+  // we care only about the very first result as question can be changed before
+  // the modal is closed
+  const isSavedQuestionChanged = useSelector(
+    getIsSavedQuestionChanged,
+    () => true,
```

Comment:
I didn't even know about this pattern until today.
As I'm reading about it, it seems that this will always re-render and I think I found that it's considered a bad practice in one place.

I have no horse in the race, and no opinion given that I'm just learning about it but I was wondering if `useRef` would be a better choice here?

I experimented with it locally and it does the work.

Reply:
no matter, will you choose useRef, useMemo or the variable outside component or this pattern, it's a way to get the very first value from selector

Reply:
nit: Maybe `const [isSavedQuestionInitiallyChanged] = useState(isSavedQuestionChanged)` to make it more obvious.

Reply:
@nemanjaglumac it's a widely used pattern, redux needs to know when to trigger component render - so redux uses the second argument passed a comparator function - it compares prev and current result and if it's changed - render